### PR TITLE
Update with nodejs functions

### PR DESCRIPTION
**Pull request by ronish31**
@idohalevi 
The changes are to:
Prevent logs from type "START", "END", "REPORT" from being sent to logz.io
Work with nodejs functions.

### CODE REVIEW

File: python3/cloudwatch/src/lambda_function.py [hunk 1]
```diff
@@ -56,18 +51,18 @@ def _extract_lambda_log_message(log, log_group):
             start_split = 0
 
         message_parts = str_message[start_split:].split('\t')
-        if len(message_parts) == 3:
+        size = len(message_parts)
+        if (size == 3 or size == 4):
             log['@timestamp'] = message_parts[0]
             log['requestID'] = message_parts[1]
-            log['message'] = message_parts[2]
+            log['message'] = message_parts[size - 1]
```
Comment (idohalevi) [inline python3/cloudwatch/src/lambda_function.py]: If `size == 4` what do we have in `message_parts[2]`?

File: python3/cloudwatch/src/lambda_function.py [hunk 2]
```diff
@@ -128,6 +121,7 @@ def lambda_handler(event, context):
             raise TypeError("Expected log inside logEvents to be a dict but found another type")
 
         _parse_cloudwatch_log(log, additional_data)
-        shipper.add(log)
+        if not log['message'].startswith('START') and not log['message'].startswith('END') and not log['message'].startswith('REPORT'):
```
Comment (idohalevi) [inline python3/cloudwatch/src/lambda_function.py]: Why here? Also, let's not use `not` it makes it hard to read the code

Comment (ronish31) [reply python3/cloudwatch/src/lambda_function.py]: please review changes on new commit
https://github.com/logzio/logzio_aws_serverless/commit/8b2689012a9bf9f925a2da9f506b514186d2e20d

File: python3/cloudwatch/src/lambda_function.py (hunk 1 above)
Comment (ronish31) [reply python3/cloudwatch/src/lambda_function.py]: The word "INFO"
nodejs_event

File: python3/cloudwatch/src/lambda_function.py [hunk 3]
```diff
@@ -117,11 +117,12 @@ def lambda_handler(event, context):
 
     logger.info("About to send {} logs".format(len(aws_logs_data['logEvents'])))
     for log in aws_logs_data['logEvents']:
-        if not isinstance(log, dict):
-            raise TypeError("Expected log inside logEvents to be a dict but found another type")
+        is_extra_data = log['message'].startswith('START') or log['message'].startswith('END') or log['message'].startswith('REPORT')
+        if not is_extra_data:
```
Comment (idohalevi) [inline python3/cloudwatch/src/lambda_function.py]: @ronish31 turn it into a function and based on the return value decide if we want to ship the log or not

**Comment by ronish31** [comment]
@idohalevi 
please review vhanges in this commit: c4fedab3d736f23436f218aca9545b719d4d9f77

File: python3/cloudwatch/tests/lambda_tests.py [hunk 4]
```diff
@@ -15,6 +15,8 @@
 # CONST
 BODY_SIZE = 10
 STRING_LEN = 10
+python_event_size = 3
+nodejs_event_size = 4
```
Comment (idohalevi) [inline python3/cloudwatch/tests/lambda_tests.py]: Capital. Also, you have some unused imports

File: python3/kinesis/src/lambda_function.py [hunk 5]
```diff
@@ -38,7 +38,7 @@ def _get_type(data):
         json_data = json.loads(data)
         return json_data["source"].split('.')[1]
     except (KeyError, ValueError):
-        return "kinesis_lambda"
+        return "logzio_kinesis_stream"
```
Comment (idohalevi) [inline python3/kinesis/src/lambda_function.py]: It's better to keep it `kinesis_lambda` for backward compatibility, even if it's not aligned with CW.

Comment (ronish31) [reply python3/kinesis/src/lambda_function.py]: Opened a PR in the docs about it @imnotashrimp

File: python3/cloudwatch/src/lambda_function.py [hunk 6]
```diff
@@ -104,20 +110,27 @@ def _get_additional_logs_data(aws_logs_data, context):
...
-        logger.info("Failed to find 'TYPE' environment variables. Using 'logzio_cloudwatch_lambda' instead")
+        logger.info("Using default TYPE 'logzio_cloudwatch_lambda'.")
         additional_data['type'] = 'logzio_cloudwatch_lambda'
-
     return additional_data
 
 
+def _is_valid_log(log):
+    try:
+        if log['message'].startswith('START') or log['message'].startswith('END') or log['message'].startswith('REPORT'):
```
[the remark this answers is not in the document]
Comment (ronish31) [reply python3/cloudwatch/src/lambda_function.py]: It was for an old case- deleted.

File: python3/cloudwatch/tests/lambda_tests.py [hunk 7]
```diff
@@ -102,6 +129,35 @@ def _validate_json_data(self, request, data, context):
...
+        self.assertEqual(body_logs['@timestamp'], original_log[0])
+        self.assertEqual(body_logs['requestID'], original_log[1])
+        if 'log_level' in body_logs:
+            self.assertEqual(body_logs['log_level'], original_log[2])
+        message_list = original_log[size_of_event-1].split(",")
+        original_message_list = body_logs['message'].split(",")
+        for i in range(len(message_list)):
+            self.assertEqual(message_list[i], original_message_list[i])
+
+    def _test_aws_event(self, event_builder, size_of_event):
```
[the remark this answers is not in the document]
Comment (ronish31) [reply python3/cloudwatch/tests/lambda_tests.py]: Refactored

**Pull request merged**